### DISCUSSION TITLE
How do I keep spinner after it's done?

### DISCUSSION THREAD

**Discussion by foodornt** [charmbracelet/gum#372] [Q&A]
`gum spin --spinner dot --title "Buying Bubble Gum..." -- sleep 5`
after 5 seconds, the spinner and the text will disappear
how do I make it persist?

**Comment by anins1der** [comment]
You can only use a longer sleep time :(

**Comment by beltagyy** [comment]
gum spin clears the spinner line when the command finishes by design.

To keep text on screen after the spin ends, you have a few options:
1. Echo something after the spin exits:
   gum spin --spinner dot --title "Buying Bubble Gum..." -- sleep 5
   echo "Done buying bubble gum!"

2. Use --show-output so any command output (or the title) is printed before the spinner clears:
   gum spin --show-output --title "Buying Bubble Gum..." -- echo "finished"

3. If you need the spinner itself to persist permanently, gum spin is not the right tool. Build a small Bubble Tea app where you control the lifecycle of the spinner model directly.
